### DISCUSSION TITLE
Oobabooga extension

### DISCUSSION THREAD

**Discussion by jsboige** [Dicklesworthstone/swiss_army_llama#1] [Ideas]
Any chance to see an [oobabooga extension](https://github.com/oobabooga/text-generation-webui/blob/main/docs/Extensions.md) adapted from this?

**Comment by Dicklesworthstone** [comment]
I think it serves a pretty different purpose than oobabooga, so not sure I really see the overlap or usefulness of combining them. Isn't Oobabooga basically a web frontend for interacting with LLMs? For this kind of functionality in my project, I think the swagger page is more convenient.

**Comment by jsboige** [comment]
Well oobabooga is more or less the automatic1111 of LLMs. It supports all kinds of models, and then offers several user interface for completions, chat, model settings, personae, and additional services through extensions.
Embeddings support are a bit lagging though and only supported through this [specific extension](https://github.com/oobabooga/text-generation-webui/tree/main/extensions/openai) that mimicks OpenAI's API.
It seems to me like your project would be a good fit to complete the official API with better dedicated embedding support just like the official API adds plenty dedicated LLama parameters to what the OpenAI clone API provides.
